tests/test_sortedqueue.py

```diff
@@ -0,0 +1,165 @@
+"""Tests for the SortedQueue class."""
+
+import unittest
+
+import numpy as np
+import pytest
+
+from laser_core.sortedqueue import SortedQueue
+
+messages = []
+
+
+class TestSortedQueue(unittest.TestCase):
+    """Tests for the SortedQueue class."""
+
+    def setUp(self):
+        # 31 41 59 26 53 58 97
+        self.pq = SortedQueue(7, np.array([31, 41, 59, 26, 53, 58, 97], dtype=np.int32))
+
+    def test_push_pop(self):
+        """Test pushing and popping elements from the sorted queue."""
+        self.pq.push(0)
+        self.pq.push(1)
+        self.pq.push(2)
+        self.pq.push(3)
+        self.pq.push(4)
+        self.pq.push(5)
+        self.pq.push(6)
+
+        assert self.pq.popiv() == (3, 26)
+        assert self.pq.popiv() == (0, 31)
+        assert self.pq.popiv() == (1, 41)
+        assert self.pq.popiv() == (4, 53)
+        assert self.pq.popiv() == (5, 58)
+        assert self.pq.popiv() == (2, 59)
+        assert self.pq.popiv() == (6, 97)
+
+    def test_push_pop_random(self):
+        """Test pushing and popping random values from the sorted queue."""
+        values = np.random.randint(0, 100, 1024, dtype=np.int32)
+        self.pq = SortedQueue(len(values), values)
+        for i in range(len(values)):
+            self.pq.push(i)
+        minimum = 0
+        while len(self.pq) > 0:
+            value = self.pq.popv()
+            assert value >= minimum
+            minimum = value
+
+    def test_peek(self):
+        """Test peeking at the top element of the sorted queue."""
+        self.pq.push(0)
+        self.pq.push(1)
+        self.pq.push(2)
+        self.pq.push(3)
+        self.pq.push(4)
+        self.pq.push(5)
+        self.pq.push(6)
+
+        assert self.pq.peekiv() == (3, 26)
```

Comment:
When testing a peek, would it be better to do two in a row to prove that the peek is just a peek and not a pop?